Revit_Core_Engine/Query/Shape.cs

```diff
@@ -68,6 +70,32 @@ public static Autodesk.Revit.DB.ConnectorProfileType Shape(this MEPCurve mEPCurv
         }
 
         /***************************************************/
+
+        [Description("Determines shape of the MEP Family Symbol based on its primary connector.")]
+        [Input("familySymbol", "Family Symbol to determine the shape of.")]
+        [Input("doc", "Document object of the Family Symbol.")]
+        [Input("settings", "Revit adapter settings.")]
+        [Output("shape", "Shape of an Family Symbol, which can be round, rectangular or oval. If the shape cannot be determined, an invalid connector shape is returned.")]
+        public static ConnectorProfileType Shape(this FamilySymbol familySymbol, Document doc, RevitSettings settings = null)
+        {
+            if (familySymbol == null || doc == null)
+            {
+                BH.Engine.Base.Compute.RecordError("Family Symbol or input document cannot be null.");
```

Comment:
Assuming that I am correct that `doc` input is not needed ⬆️ 
```suggestion
                BH.Engine.Base.Compute.RecordError("Querying MEP shape of an element failed because the queried family symbol cannot be null.");
```